Python/thread_foobar.h

```diff
@@ -123,10 +126,74 @@ PyThread_delete_key_value(int key)
 
 }
 
+
 void
 PyThread_ReInitTLS(void)
 {
 
 }
 
+
+/* Thread Specific Storage (TSS) API */
+
+int
+PyThread_tss_create(Py_tss_t *key)
+{
+    assert(key != NULL);
+    /* If the key has been created, function is silently skipped. */
+    if (key->_is_initialized)
+        return 0;
+
+    Py_tss_t new_key;
+    /* A failure in this case returns -1 */
+    if (!new_key._key)
+        return -1;
+    key->_key = new_key._key;
+    key->_is_initialized = true;
+    return 0;
+}
+
+void
+PyThread_tss_delete(Py_tss_t *key)
+{
+    assert(key != NULL);
+    /* If the key has not been created, function is silently skipped. */
+    if (!key->_is_initialized)
+        return;
+
+    key->_is_initialized = false;
+}
+
+int
+PyThread_tss_set(Py_tss_t key, void *value)
+{
+    int ok;
+
+    /* A failure in this case returns -1 */
+    if (!ok)
+        return -1;
+    return 0;
+}
+
+void *
+PyThread_tss_get(Py_tss_t key)
+{
+    void *result;
+
+    return result;
+}
+
+void
+PyThread_tss_delete_value(Py_tss_t key)
+{
```

Comment:
I had assumed it was a dummy implementation for testing / demonstration purposes.  It might be an easier way to understand the requirements of the threading API without being bogged down with the details of specific implementations.  That said, I don't know how useful that is in practice. I don't think I ever looked at it, opting instead to look at the pthread implementation since pthreads are already familiar...